crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs

```diff
@@ -126,7 +126,7 @@ pub(crate) fn used_dummy_variable(checker: &Checker, binding: &Binding, binding_
     // autofixing the diagnostic for assignments. See:
     // - <https://github.com/astral-sh/ruff/issues/14790>
     // - <https://github.com/astral-sh/ruff/issues/14799>
-    if !binding.kind.is_assignment() {
+    if !binding.kind.is_assignment() && !matches!(binding.kind, BindingKind::LoopVar) {
```

Comment:
If we're already using `matches!` we might as well combine these checks:


```suggestion
    if !matches!(binding.kind, BindingKind::Assignment | BindingKind::Assignment) {
```